Specify that range patterns must be nonempty.

theemathas opened the pull request:
This changes the reference to match the behavior of the compiler, based on my empirical testing.

See also https://github.com/rust-lang/rust/issues/149165

theemathas committed:
Specify that range patterns must be nonempty.

This changes the reference to match the behavior of the compiler, based on my empirical testing.

See also https://github.com/rust-lang/rust/issues/149165

ehuss commented on src/patterns.md:
```diff
@@ -554,10 +554,12 @@ It is written as `..=` followed by the upper bound.
-That is, in `a..=b`, a &le; b must be the case.
-For example, it is an error to have a range pattern `10..=0`.
+r[patterns.range.constraint-nonempty]
+A range pattern must match at least one possible value. In other words:
```
The wording here seemed little strange to me. Does this make sense? Obviously range patterns that are outside the range of the scrutinee won't match the value at all.

```suggestion
A range pattern must be able to match at least one possible value. In other words:
```

theemathas replied:
I wrote it this way to be consistent with the other lines in this section, for example:

> The exclusive range pattern matches all values from the lower bound up to, but not including the upper bound.

traviscross committed:
Revise wording on `constraint-nonempty`

Let's revise the wording to avoid a possible ambiguity about what
we're matching against.
